mitodl/open-discussions
# Make contributors API moderator-only and add moderator-only serializer for contributors

#### What are the relevant tickets?
Part of #864 

#### What's this PR do?
 - Make contributors list view accessible only to moderators, and adds full name and email to the information shown. At the moment the UI does not use the contributors API at all so this should not cause any issues.
 - Remove contributor detail GET view since there is no use case for it.
 - Add logic to detect if a user is a moderator of the channel in order to determine whether to show the email and name of moderators. For non-moderators only the username is shown.
 - Remove moderator detail GET view since there is no use case for it.
- Add code to `Api.is_moderator` to verify that the given username is not falsey. Reddit interprets a falsey username as a request for all moderator information.
 - Add explicit check for anonymous users in `is_moderator` to fix reddit issue.

#### How should this be manually tested?
Note that none of these changes should cause any user facing behavior changes.

Log in as a contributor and view `/api/v0/channels/<name>/moderators/`. You should see a list of dicts with only the username.

Log in as a moderator and view the same URL. You should now see full names and email addresses.
